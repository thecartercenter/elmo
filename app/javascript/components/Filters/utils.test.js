import {
  getButtonHintString,
  getItemNameFromId,
  getFilterString,
  submitSearch,
  isQueryParamTruthy,
} from './utils';
import { getEmptySubmitterTypeMap } from './model';

import { getFiltersStore } from './testUtils';

it('gets hints (0)', () => {
  const result = getButtonHintString([]);
  expect(result).toMatchSnapshot();
});

it('gets hints (small number)', () => {
  const result = getButtonHintString(['one']);
  expect(result).toMatchSnapshot();
});

it('gets hints (too many)', () => {
  const result = getButtonHintString(['one', 'two', 'three']);
  expect(result).toMatchSnapshot();
});

it('gets item name (found)', () => {
  const result = getItemNameFromId([{ id: '1', name: 'One' }], '1');
  expect(result).toMatchSnapshot();
});

it('gets item name (not found)', () => {
  const result = getItemNameFromId([], '1');
  expect(result).toMatchSnapshot();
});

it('gets filter string (no filters)', () => {
  const filtersStore = getFiltersStore();
  const emptyFilters = {
    ...filtersStore,
    selectedFormIds: [],
    conditionSetStore: {
      ...filtersStore.conditionSetStore,
      conditions: [],
    },
    isReviewed: null,
    selectedSubmittersForType: getEmptySubmitterTypeMap(),
    advancedSearchText: '',
  };

  const result = getFilterString(emptyFilters);
  expect(result).toMatchSnapshot();
});

it('gets filter string (all filters)', () => {
  const populatedFilters = {
    ...getFiltersStore(),
    selectedFormIds: ['1', '3'],
    advancedSearchText: 'query',
  };

  const result = getFilterString(populatedFilters);
  expect(result).toMatchSnapshot();
});

it('submits searches', () => {
  // Page should go away after search, but other params should pass through.
  window.location.search = '?foo=bar&page=2';
  expect(window.location.assign.mock.calls).toMatchSnapshot();

  submitSearch('foo');
  expect(window.location.assign.mock.calls).toMatchSnapshot();

  window.location.assign.mockClear();
  submitSearch(null);
  expect(window.location.assign.mock.calls).toMatchSnapshot();

  window.location.assign.mockClear();
  window.location.search = '?';
  submitSearch(null);
  expect(window.location.assign.mock.calls).toMatchSnapshot();
});

it('checks if param is truthy', () => {
  window.location.search = '?foo=&bar=baz';
  expect(isQueryParamTruthy('foo')).toMatchSnapshot();
  expect(isQueryParamTruthy('bar')).toMatchSnapshot();
  expect(isQueryParamTruthy('baz')).toMatchSnapshot();
});
