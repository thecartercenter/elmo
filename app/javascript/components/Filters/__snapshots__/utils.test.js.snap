// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if param is truthy 1`] = `false`;

exports[`checks if param is truthy 2`] = `true`;

exports[`checks if param is truthy 3`] = `false`;

exports[`gets filter string (all filters) 1`] = `"form-id:(1|3) {One}:\\"5\\" reviewed:1 submitter-id:(B) group-id:(B) query"`;

exports[`gets filter string (no filters) 1`] = `""`;

exports[`gets hints (0) 1`] = `""`;

exports[`gets hints (small number) 1`] = `" (one)"`;

exports[`gets hints (too many) 1`] = `" (3 filters)"`;

exports[`gets item name (found) 1`] = `"One"`;

exports[`gets item name (not found) 1`] = `"Unknown"`;

exports[`submits searches 1`] = `Array []`;

exports[`submits searches 2`] = `
Array [
  Array [
    "?foo=bar&search=foo",
  ],
]
`;

exports[`submits searches 3`] = `
Array [
  Array [
    "?foo=bar",
  ],
]
`;

exports[`submits searches 4`] = `
Array [
  Array [
    "/pathname",
  ],
]
`;
